lighthouse-core/config/config.js

```diff
@@ -193,13 +193,17 @@ function assertValidAudit(auditDefinition, auditPath) {
   if (typeof auditDefinition.meta.failureDescription !== 'string' &&
     auditDefinition.meta.informative !== true &&
     auditDefinition.meta.scoringMode !== Audit.SCORING_MODES.NUMERIC) {
-    log.warn('config', `${auditName} has no failureDescription and should.`);
+    throw new Error(`${auditName} has no failureDescription and should.`);
   }
 
   if (typeof auditDefinition.meta.helpText !== 'string') {
     throw new Error(
       `${auditName} has no meta.helpText property, or the property is not a string.`
     );
+  } else if (auditDefinition.meta.helpText === '') {
+    throw new Error(
```

Comment:
> worth noting this might break some custom audits that had no need for helpText since they didn't generate an HTML report, which is unfortunate

yeah, I started the trend by checking that everything had a string `helpText` (though `''` was acceptable), but it does kind of suck that even if you're doing output for CI only you still need a `helpText`, a `description` and a `fallbackDescription`.

- We could switch back to the log warning @paulirish had added before for `fallbackDescription`, but it is handy to have a real error so that tests fail when adding a new audit
- or we could pipe in the report output type and do extra audit validation for the html report or move this validation to somewhere in the report generator
- or we could add some of these checks to our unit tests so that the things all audits need are still checked here with an error and everything else is a warning (so custom audits get a hint of how they should work if they want to work with an html report). We have something like this in [`runner-test.js`](https://github.com/GoogleChrome/lighthouse/blob/7c461bf0de546f6939ee961de5e215ef694ebf0c/lighthouse-core/test/runner-test.js#L359-L367) where we check that all core audits filename and ID match.